src/components/QRCodeScanner/CameraView.js

```diff
@@ -44,14 +53,83 @@ const HeaderWrapper = styled.SafeAreaView`
   width: 100%;
 `;
 
+const ButtonWrapper = styled.View`
+  width: 100%;
+  position: absolute;
+  bottom: 60px;
+  justify-content: center;
+  align-items: center;
+`;
+
 type Props = {
   onQRRead: (barcode: Barcode) => void,
   onCancel: () => void,
   rectangleColor: string,
   rectangleSize: number,
 };
 
-export class CameraView extends PureComponent<Props> {
+type State = {
+  isLoading: boolean,
+}
+
+const ERROR_TIMEOUT = 15000;
+
+export default class CameraView extends React.Component<Props, State> {
+  state = {
+    isLoading: false,
+  }
+
+  timeout: TimeoutID;
+
+  componentWillUnmount() {
+    if (this.timeout) {
+      clearTimeout(this.timeout);
+    }
+  }
+
+  handleError = () => {
+    Toast.show({
+      message: 'Cannot scan QR code',
+      type: 'warning',
+      autoClose: true,
+    });
+    this.props.onCancel();
+  }
+
+  handleGalleryPress = () => {
+    this.setState({ isLoading: true });
+    ImagePicker.openPicker({
+      includeBase64: true,
+    })
+      .then(image => {
+        this.timeout = setTimeout(() => {
+          this.handleError();
+        }, ERROR_TIMEOUT);
+        const buffer = Buffer.from(image.data, 'base64');
+        Jimp.read(buffer, (e, parsedImg) => {
```

Comment:
This takes ridiculous amount of time on my device (android), maybe because I take large images with my camera by default. I mean it took several minutes already and I'm still waiting :D

Reply:
Yeah, that's what the timeout is for... guess it doesn't work as it should then :/ will try to reproduce and fix